<launch>
  <arg name="data_dir" default = "/root/shared/"/>
  <arg name="res_dir" default = "/root/shared/"/>
  <arg name="ts_length" default="150"/> 
  <arg name="dt" default="0.1"/> 
  <arg name="subsampling" default="false"/> 
  <arg name="addnoise" default="false"/> 
  <arg name="std" default="0.05"/>
  <arg name="cd_method" default="fpcmci"/>
  <arg name="sig_alpha" default="0.05"/>
  <arg name="filter_alpha" default="0.05"/>
  <arg name="min_lag" default="1"/>
  <arg name="max_lag" default="1"/>

  <node name="mytiago_data_collector" pkg="mytiago_causal_discovery" type="collecting_data.py" output="screen">
    <param name="data_dir"   value="$(arg data_dir)"/> 
    <param name="ts_length"   value="$(arg ts_length)"/> 
    <param name="dt"   value="$(arg dt)"/> 
    <param name="subsampling"  value="$(arg subsampling)"/>
    <param name="addnoise"  value="$(arg addnoise)"/>
    <param name="std"  value="$(arg std)"/>
  </node>

  <node name="mytiago_causal_discovery" pkg="mytiago_causal_discovery" type="causal_discovery.py" output="screen">
    <param name="data_dir"   value="$(arg data_dir)"/> 
    <param name="res_dir"   value="$(arg res_dir)"/> 
    <param name="cd_method"   value="$(arg cd_method)"/> 
    <param name="sig_alpha"   value="$(arg sig_alpha)"/>
    <param name="filter_alpha"   value="$(arg filter_alpha)"/>
    <param name="min_lag"   value="$(arg min_lag)"/>
    <param name="max_lag"   value="$(arg max_lag)"/>
  </node>
  
</launch>